src/components/NamePicker.tsx

```diff
@@ -1,10 +1,76 @@
-import React from 'react';
-import TextField from '@material-ui/core/TextField';
+import React, { useState, MouseEvent, ChangeEvent } from 'react';
 import Grid from '@material-ui/core/Grid';
+import Popover from '@material-ui/core/Popover';
+import TextField from '@material-ui/core/TextField';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
 import { makeStyles } from '@material-ui/core/styles';
 
 export default function NamePicker(): JSX.Element {
   const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [anchorEl2, setAnchorEl2] = useState<null | HTMLElement>(null);
+  const [disabled1, setDisabled1] = useState(true);
+  const [disabled2, setDisabled2] = useState(true);
+  const [firstName, setFirstName] = useState({
+    firstSounds: false,
+    firstSimilar: false,
+    firstInitials: false,
+  });
+
+  const [lastName, setLastName] = useState({
+    lastSounds: false,
+    lastSimilar: false,
+    lastInitials: false,
+  });
+
+  const handleClick1 = (event: MouseEvent<HTMLElement>): void => {
+    setAnchorEl(event.currentTarget);
+    if (anchorEl === null) {
+      setDisabled1(!disabled1);
+      setFirstName({
+        firstSounds: false,
+        firstSimilar: false,
+        firstInitials: false,
+      });
+    }
+  };
+
+  const handleClick2 = (event: MouseEvent<HTMLElement>): void => {
+    setAnchorEl2(event.currentTarget);
+    if (anchorEl2 === null) {
+      setDisabled2(!disabled2);
+      setLastName({
+        lastSounds: false,
+        lastSimilar: false,
+        lastInitials: false,
+      });
+    }
+  };
+
+  const handleClose1 = (): void => {
+    setAnchorEl(null);
+  };
+  const handleClose2 = (): void => {
+    setAnchorEl2(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const open2 = Boolean(anchorEl2);
+  const fName = open ? 'simple-popover' : undefined;
+  const lName = open ? 'simple-popover' : undefined;
```

Comment:
Is this a bug? Should it be `open2` here?

Note that if you use the `!!` trick, the code is quite readable and you avoid the local variable:
`const fName = !!anchorEl ? ....`
Actually, since you are just checking for falsy or truthy, you don't even need `!!`:
`const fName = anchorEl ? 'simple-popover' : undefined;`
And for another refinement, since you are returning `undefined`, you can use short-circuit boolean expressions:
`const fName = anchorEl && 'simple-popover';`
If `anchorEl` is `undefined`, the left-hand side evaluates to false and returns itself as the value of the entire expression, so it sets `fName` to `undefined`. If it is defined, though, then the right-hand side evaluates and returns itself as the entire expression, so it sets `fName` to `'simple-popover'. This final form is very common in JS and TS.

Reply:
Sorry was a bug that has been fixed and have refactored fName and lName. && for this is so much better-I have to do better to remember this, thanks!